guides/create_an_app/4-query.md

```diff
@@ -1,5 +1,88 @@
 # 4. Query
 
-Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin nibh augue, suscipit a, scelerisque sed, lacinia in, mi. Cras vel lorem. Etiam pellentesque aliquet tellus. Phasellus pharetra nulla ac diam. Quisque semper justo at risus. Donec venenatis, turpis vel hendrerit interdum, dui ligula ultricies purus, sed posuere libero dui id orci. Nam congue, pede vitae dapibus aliquet, elit magna vulputate arcu, vel tempus metus leo non est. Etiam sit amet lectus quis est congue mollis. Phasellus congue lacus eget neque. Phasellus ornare, ante vitae consectetuer consequat, purus sapien ultricies dolor, et mollis pede metus eget nisi. Praesent sodales velit quis augue. Cras suscipit, urna at aliquam rhoncus, urna quam viverra nisi, in interdum massa nibh nec erat.
+## Retrieving from database
+
+Everytime when you refresh your page, you lose the message history and that is not ideal if you want to know what has been discussed. We have all these stored messages in our database, and we want to retrieve it when someone logs in. Therefore, we will need to retrieve these messages and display them in the chat dialog.
+
+## Perform a callback `after_join`.
+Go to the join function, and add the line just below `if authorized?(payload) do`
+
+We are going to send a message contaning our self *PID(Process Identifier)* as the first parameter. The second parameter is a message to run a function called `after_join`. This function is identified by an atom type.
```

Comment:
I don't understand this line: "We are going to send a message containing our SELF(????) ... as the first parameter."